internal/cloud/update_uploader.go

```diff
@@ -0,0 +1,128 @@
+package cloud
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/duration"
+	"github.com/golang/protobuf/ptypes/timestamp"
+
+	"github.com/windmilleng/tilt/internal/feature"
+	"github.com/windmilleng/tilt/internal/store"
+)
+
+type UpdateUploader struct {
+	client HttpClient
+	addr   Address
+
+	lastCompletedBuildCount int
+	lastFinishTime          time.Time
+}
+
+func NewUpdateUploader(client HttpClient, addr Address) *UpdateUploader {
+	return &UpdateUploader{
+		client: client,
+		addr:   addr,
+	}
+}
+
+// TODO(nick): Generate these with protobufs.
+type updateServiceSpec struct {
+	Name string `json:"name"`
+}
+
+type update struct {
+	Service      updateServiceSpec    `json:"service"`
+	StartTime    *timestamp.Timestamp `json:"start_time"`
+	Duration     *duration.Duration   `json:"duration"`
+	IsLiveUpdate bool                 `json:"is_live_update"`
+
+	// 0 = SUCCESS, 1 = FAILURE
+	Result            int    `json:"result"`
+	ResultDescription string `json:"result_description"`
+
+	// TODO(nick): auto-create Snapshot IDs?
+}
+
+// Check the engine state to see if we have any updates.
+func (u *UpdateUploader) makeUpdates(st store.RStore) []update {
+	state := st.RLockState()
+	defer st.RUnlockState()
+
+	// Check if this feature is disabled
+	if !state.Features[feature.UpdateHistory] {
+		return nil
+	}
+
```

Comment:
let's also return `if state.TeamName == ""`

Reply:
oh, we're not supporting update-history for non-teams?